src/test/javascript/spec/component/course/course-management-exercises-search.component.spec.ts

```diff
@@ -0,0 +1,139 @@
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [
+                CourseManagementExercisesSearchComponent,
+                MockPipe(ArtemisTranslatePipe),
+                MockComponent(CourseExerciseCardComponent),
+                MockDirective(TranslateDirective),
+                MockDirective(NgModel),
+            ],
+            providers: [],
+        }).compileComponents();
+        fixture = TestBed.createComponent(CourseManagementExercisesSearchComponent);
+        comp = fixture.componentInstance;
+        exercises = [exercise1, exercise2, exercise3];
+    });
+
+    it('should initialize', () => {
+        fixture.detectChanges();
+        expect(CourseManagementExercisesSearchComponent).toBeDefined();
+    });
+
+    it('should have empty filter on init', () => {
+        comp.ngOnInit();
+        expect(comp.exerciseNameSearch).toEqual('');
+        expect(comp.exerciseCategorySearch).toEqual('');
+        expect(comp.exerciseTypeSearch).toEqual('all');
+    });
+
+    it('should change filter on name change', () => {
+        const filter = new ExerciseFilter();
+        filter.exerciseNameSearch = 'test';
+        const emitSpy = jest.spyOn(comp.exerciseFilter, 'emit');
+        fixture.detectChanges();
+        const button = fixture.debugElement.nativeElement.querySelector('#saveFilterButton');
+        comp.exerciseNameSearch = filter.exerciseNameSearch;
+        button.click();
+        expect(emitSpy).toHaveBeenCalled();
+        expect(emitSpy).toHaveBeenCalledWith(filter);
+    });
+
+    it('should change filter on category change', () => {
+        const filter = new ExerciseFilter();
+        filter.exerciseCategorySearch = 'homework';
+        const emitSpy = jest.spyOn(comp.exerciseFilter, 'emit');
+        fixture.detectChanges();
+        const button = fixture.debugElement.nativeElement.querySelector('#saveFilterButton');
+        comp.exerciseCategorySearch = filter.exerciseCategorySearch;
+        button.click();
+        expect(emitSpy).toHaveBeenCalled();
+        expect(emitSpy).toHaveBeenCalledWith(filter);
+    });
+
+    it('should change filter on type change', () => {
+        const filter = new ExerciseFilter();
+        filter.exerciseTypeSearch = 'programming';
+        const emitSpy = jest.spyOn(comp.exerciseFilter, 'emit');
+        fixture.detectChanges();
+        const button = fixture.debugElement.nativeElement.querySelector('#saveFilterButton');
+        comp.exerciseTypeSearch = filter.exerciseTypeSearch;
+        button.click();
+        expect(emitSpy).toHaveBeenCalled();
+        expect(emitSpy).toHaveBeenCalledWith(filter);
+    });
+
+    describe('Exercise Filter Test', () => {
+        it('should be empty on create', () => {
+            const filter = new ExerciseFilter();
+            expect(filter.isEmpty()).toEqual(true);
```

Comment:
```suggestion
            expect(filter.isEmpty()).toBe(true);
```
Please replace other occurrences